### PR TITLE
Improve developer-facing evaluations API

### PR DESCRIPTION
**Pull request by dqbd**
Fixes # (issue)

### CODE REVIEW

File: langchain-core/src/runnables/base.ts [hunk 1]
```diff
@@ -2364,3 +2365,47 @@ export class RunnablePick<
     return IterableReadableStream.fromAsyncGenerator(wrappedGenerator);
   }
 }
+
+type RunnableTraceeableFunc<RunInput, RunOutput> = (
```
Comment (jacoblee93) [inline langchain-core/src/runnables/base.ts:2369]: Small typo

File: langchain/package.json [hunk 2]
```diff
@@ -1513,7 +1513,7 @@
     "js-yaml": "^4.1.0",
     "jsonpointer": "^5.0.1",
     "langchainhub": "~0.0.6",
-    "langsmith": "~0.0.59",
+    "langsmith": "~0.1.0",
```
Comment (jacoblee93) [inline langchain/package.json:1516]: Can you bump to 0.1.1?

**Comment by jacoblee93** [comment]
Overall LGTM - could you update docs quick?

I also did a quick scan and didn't see `smith/index.ts` or `runner_utils.ts` imported anywhere else but would be good to triple check just in case.

**Comment by jacoblee93** [comment]
Thank you!

File: docs/core_docs/docs/guides/langsmith_evaluation.mdx [hunk 3]
```diff
@@ -194,31 +194,15 @@ const notUnsure = async ({
...
-          input: rawInput.input,
-          prediction: rawPrediction.output,
-          reference: rawReferenceOutput.output,
-        };
-      },
-    },
-  ],
+const evaluators: RunEvalConfig["evaluators"] = [
+  // LangChain's built-in evaluators
+  LabeledCriteria("correctness"),
```
Comment (agola11) [inline docs/core_docs/docs/guides/langsmith_evaluation.mdx:199]: need to import these in example

**Pull request merged**